File: app/src/main/java/com/moducode/gw2serveralarm/data/ServerModel.java

package com.moducode.gw2serveralarm.data;

import android.graphics.Color;
import android.support.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.Comparator;

public class ServerModel implements Comparable<ServerModel> {

    @SerializedName("id")
    @Expose
    private int id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("population")
    @Expose
    private String population;

    private int populationLevel;

    public ServerModel(int id, String name, String population) {
        this.id = id;
        this.name = name;
        this.population = population;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPopulation() {
        return population;
    }

    public void setPopulation(String population) {
        this.population = population;
    }

    public void setPopulationLevel(int level){
        this.populationLevel = level;
    }

    public int getPopulationLevel(){
        switch (population){
            case "Low":
                return 0;
            case "Medium":
                return 1;
            case "High":
                return 2;
            case "VeryHigh":
                return 3;
            case "Full":
                return 4;
            default:
                return 0;
        }
    }

    public int getPopulationColor(){
        switch (population){
            case "Low":
                return Color.parseColor("#C6E0B4");
            case "Medium":
                return Color.parseColor("#A9D08E");
            case "High":
                return Color.parseColor("#FFC000");
            case "VeryHigh":
                return Color.parseColor("#ED7D31");
            case "Full":
                return Color.parseColor("#FF0000");
            default:
                return Color.parseColor("#C6E0B4");
        }
    }

    public String getIdString(){
        return String.valueOf(this.id);
    }

    @Override
    public int compareTo(@NonNull ServerModel serverModel) {
        return serverModel.getPopulationLevel() - this.getPopulationLevel();
    }
}

File: app/src/main/java/com/moducode/gw2serveralarm/data/ServerModel.kt
package com.moducode.gw2serveralarm.data

import android.graphics.Color

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class ServerModel(@field:SerializedName("id")
                  @field:Expose
                  val id: Int,
                  @field:SerializedName("name")
                  @field:Expose
                  val name: String,
                  @field:SerializedName("population")
                  @field:Expose
                  val population: String) : Comparable<ServerModel> {

    private val populationLevel: Int
        get() {
            return when (population) {
                "Low" -> 0
                "Medium" -> 1
                "High" -> 2
                "VeryHigh" -> 3
                "Full" -> 4
                else -> 0
            }
        }

    val populationColor: Int
        get() {
            return when (population) {
                "Low" -> Color.parseColor("#C6E0B4")
                "Medium" -> Color.parseColor("#A9D08E")
                "High" -> Color.parseColor("#FFC000")
                "VeryHigh" -> Color.parseColor("#ED7D31")
                "Full" -> Color.parseColor("#FF0000")
                else -> Color.parseColor("#C6E0B4")
            }
        }

    val idString: String
        get() = this.id.toString()

    override fun compareTo(other: ServerModel): Int {
        return other.populationLevel - this.populationLevel
    }

    companion object {
        val fakeServer: ServerModel
            get() = ServerModel(6969, "TestServer", "null")
    }
}
